testng/src/main/java/io/cucumber/testng/TestNGCucumberRunner.java

```diff
@@ -150,12 +152,11 @@ public void finish() {
     }
 
     private List<Feature> getFeatures() {
```

Comment:
Documenting that this method is idempotent would be nice.

Reply:
A test would be nice too.

Reply:
I have a hard time with coming up with a test for this particular method since it's a private one. I added a test which demonstrates whole solution (multiple data providers and filtering scenarios)

Reply:
That's good too. It will help document stuff.